tachyon/zk/plonk/examples/circuit_test_type_traits.h

```diff
@@ -0,0 +1,48 @@
+#ifndef TACHYON_ZK_PLONK_EXAMPLES_CIRCUIT_TEST_TYPE_TRAITS_H_
+#define TACHYON_ZK_PLONK_EXAMPLES_CIRCUIT_TEST_TYPE_TRAITS_H_
+
+#include <type_traits>
+
+#include "tachyon/math/elliptic_curves/bn/bn254/bn254.h"
+#include "tachyon/zk/base/commitments/gwc_extension.h"
+#include "tachyon/zk/base/commitments/shplonk_extension.h"
+#include "tachyon/zk/lookup/halo2/scheme.h"
+#include "tachyon/zk/plonk/layout/floor_planner/simple_floor_planner.h"
+#include "tachyon/zk/plonk/layout/floor_planner/v1/v1_floor_planner.h"
+
+namespace tachyon::zk::plonk::halo2 {
+
+namespace {
+
+using SHPlonk =
+    SHPlonkExtension<math::bn254::BN254Curve, kMaxDegree, kMaxExtendedDegree,
+                     math::bn254::G1AffinePoint>;
+
+using GWC = GWCExtension<math::bn254::BN254Curve, kMaxDegree,
+                         kMaxExtendedDegree, math::bn254::G1AffinePoint>;
+using Halo2LS =
+    lookup::halo2::Scheme<typename SHPlonk::Poly, typename SHPlonk::Evals,
+                          typename SHPlonk::Commitment>;
```

Comment:
These types are being declared multiple times in the circuit test code. Since they are frequently used, how about declaring them in an appropriate place just once and reusing them?

Additionally, since `SHPlonk` and `GWC` are restricted to `BN254`, I recommend renaming them to `BN254SHPlonk` and `BN254GWC`.